src/dvc_studio_client/post_live_metrics.py

```diff
@@ -22,26 +20,23 @@
 logger.setLevel(getenv(DVC_STUDIO_CLIENT_LOGLEVEL, "INFO").upper())
 
 
-def _get_remote_url(git_repo):
-    return git_repo.git.ls_remote("--get-url")
+def _get_remote_url() -> str:
+    from dulwich.porcelain import get_remote_repo
+    from dulwich.repo import Repo
+
+    with Repo.discover() as repo:
+        _remote, url = get_remote_repo(repo)
+        return url
 
 
 @lru_cache(maxsize=1)
 def get_studio_repo_url() -> Optional[str]:
-    studio_url = None
+    from dulwich.errors import NotGitRepository
+
     try:
-        git_repo = Repo()
-        studio_url = _get_remote_url(git_repo)
-    except GitError:
-        logger.debug("Tried to find remote url for the active branch but failed.\n")
-    finally:
-        if not studio_url:
-            logger.warning(
```

Comment:
Why remove the warnings?

Reply:
There are only few places where this can happen:
1. If the `repo` is not a git repository.
2. IndexError: where the head is detached.
3. If the git repo has no remotes.

But these cases feel like exceptional to me, there is very less chance of failing in other cases. So I did not find it worth it to add. But no strong opinion.

Reply:
Brought back the warning.